plugins/techdocs-module-addons-contrib/src/ReportIssue/ReportIssueContent.tsx

```diff
@@ -0,0 +1,131 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { makeStyles, Portal, Paper } from '@material-ui/core';
+import { useGitTemplate } from './hooks';
+import {
+  PAGE_MAIN_CONTENT_SELECTOR,
+  PAGE_FEEDBACK_LINK_SELECTOR,
+  ADDON_FEEDBACK_CONTAINER_ID,
+  ADDON_FEEDBACK_CONTAINER_SELECTOR,
+} from './constants';
+import { IssueLink } from './IssueLink';
+import {
+  useShadowRootElements,
+  useShadowRootSelection,
+} from '@backstage/plugin-techdocs-react';
+import { Repository } from './types';
+import { ReportIssueProps } from './ReportIssue';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    transform: 'translate(-100%, -100%)',
+    position: 'absolute',
+    padding: theme.spacing(1),
+    zIndex: theme.zIndex.tooltip,
+    background: theme.palette.common.white,
+  },
+}));
+
+type Style = {
+  top: string;
+  left: string;
+};
+
+// Props customizing the <ReportIssue /> Addon.
+export type ReportIssueAddonContentProps = ReportIssueProps & {
+  repository: Repository;
+};
+
+export const ReportIssueAddonContent = ({
+  debounceTime,
+  templateBuilder,
+  repository,
+}: ReportIssueAddonContentProps) => {
+  const classes = useStyles();
+  const [style, setStyle] = useState<Style>();
+  const defaultTemplate = useGitTemplate(debounceTime);
+  const selection = useShadowRootSelection(debounceTime);
+
+  const [mainContent, feedbackLink] = useShadowRootElements([
+    PAGE_MAIN_CONTENT_SELECTOR,
+    PAGE_FEEDBACK_LINK_SELECTOR,
+  ]);
+
+  let [feedbackContainer] = useShadowRootElements([
+    ADDON_FEEDBACK_CONTAINER_SELECTOR,
+  ]);
+
+  if (feedbackLink) {
+    feedbackLink.style.display = 'none';
+  }
+
+  useEffect(() => {
+    if (
+      !selection ||
+      !selection.containsNode(mainContent, true) ||
+      selection?.containsNode(feedbackContainer, true)
+    ) {
+      return;
+    }
+
+    const mainContentPosition = mainContent.getBoundingClientRect();
+    const selectionPosition = selection.getRangeAt(0).getBoundingClientRect();
+
+    const distanceFromTop = selectionPosition.top - mainContentPosition.top;
+    const minDistanceFromTop = 50;
+
+    let top = distanceFromTop < minDistanceFromTop ? 101 : distanceFromTop - 16;
+
+    if (mainContentPosition.top < 0) {
+      const absMainContentTop = Math.abs(mainContentPosition.top);
+      if (distanceFromTop - absMainContentTop < minDistanceFromTop) {
+        top += 89;
+      }
+    }
+
+    setStyle({
+      top: `${top}px`,
+      left: `${selectionPosition.left + selectionPosition.width / 2}px`,
+    });
+  }, [selection, mainContent, feedbackContainer]);
+
+  if (!selection) return null;
+
+  if (!feedbackContainer) {
+    feedbackContainer = document.createElement('div');
+    feedbackContainer.setAttribute('id', ADDON_FEEDBACK_CONTAINER_ID);
+    mainContent!.prepend(feedbackContainer);
+  }
```

Comment:
Opportunistic suggestion to fix some issue kind of related:

```suggestion
  if (!feedbackContainer) {
    feedbackContainer = document.createElement('div');
    feedbackContainer.setAttribute('id', ADDON_FEEDBACK_CONTAINER_ID);
    mainContent!.prepend(feedbackContainer);
  }

  if (
    !selection ||
    !selection.containsNode(mainContent, true) ||
    selection.containsNode(feedbackContainer, true)
  ) {
    return null;
  }
```

We want to early return when there's no selection or the selected text isn't in the mainContent area, just like in the useEffect.

This avoids displaying the component when we select something in the navbars:

![image](https://github.com/user-attachments/assets/62af02d1-0921-4541-ae26-4fb145ba6386)

It also fixes a bug if you select something in the content, then something in the nav bar, the tooltip appears at the "old" location

![image](https://github.com/user-attachments/assets/fae790c2-e54c-4885-9a6f-c03d18c52713)